doc/SAI_failure_handling/dump_on_sai_failure.md

```diff
@@ -0,0 +1,91 @@
+# Dump on SAI failure #
+
+## Table of Content
+
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [High-Level Design](#high-level-design)
+- [SAI API Requirements](#sai-api-requirements)
+- [Configuration and management ](#configuration-and-management)
+    - [Config command](#config-command)
+    - [Show command](#show-command)
+    - [DB Migrator](#db-migrator)
+- [YANG Model changes](#yang-model-changes)
+- [Warmboot and Fastboot Considerations](#warmboot-and-fastboot-considerations)
+- [Testing Design](#testing-design)
+    - [Unit Tests](#unit-tests)
+    - [System tests](#system-tests)
+
+
+### Revision
+
+ | Rev |     Date    |       Author        | Change Description                         |
+ |:---:|:-----------:|:-------------------:|--------------------------------------------|
+ | 0.1 |             |      Sudharsan      | Initial version                            |
+
+### Scope
+The scope of this document is to design the handling of taking a dump during a SAI failure. 
+
+### Definitions/Abbreviations
+ 
+
+### Overview
+In the existing design, when SAI failure occurs, the orchagent aborts and all the dependent services including syncd restart. This results in failure to take the SAI, SDK and lower layer state during the problem state resulting in loss of information to debug.
+To solve this issue, whenever there is a SAI failure, orchagent requests syncd to take relevant dumps and once done, it proceeds for the abort.
```

Comment:
Are the contents of dumps at various layers, implementation/platform specific or is there a plan to formalize these ?

Reply:
That needs to be a different feature. For now we provide flexibility for the vendors to include their scripts. If this needs to be formalized, it needs to be take in the SAI forum